### PR TITLE
chore: kyasshu cap integration

### PR DESCRIPTION
**Pull request by b0xtch**

### CODE REVIEW

**Review by heldrida — CHANGES_REQUESTED** [review]
@b0xtch I saw your message, as "do not merge" so I assume it's a WIP, which I agree, there are a few things to do, so @aleacevedo approval should be ignored:

- NODE_ENV Environment should determine the cached hostname, as currently defaults to `dev`
- Documentation needs to be updated, this is mandatory as a user should be able to understand what `localhost:3000/dev` means, regardless of their knowledge about kyashu, capjs, etc
- Having the documentation/guide on that, will allow others to test this correctly (caching) before approving
- Noticed the change on Promise `createActor`, as it's not required, after I have refactored the initial proposed version into this latest class structure moved the signature which declared a promised, which you spotted as not necessary. This means that it should be refactored, as that initialisation was written to accommodate that and can now be simplified. This is probably best to do separately, let know if I should work on that as I'm integrating cap-js in cap-explorer might be quicker

File: README.md [hunk 1]
```diff
@@ -455,6 +455,56 @@ console.log(userTxns);
 ### `capRoot.time([options])`
 > ToDo
 
+### Kyasshu Layer
+
+### `capRoot.get_all_user_transactions(userId, LastEvaluatedKey)`
```
[the remark this answers is not in the document]
Comment (b0xtch) [reply README.md]: 5c1aba4

**Comment by b0xtch** [comment]
> @b0xtch I saw your message, as "do not merge" so I assume it's a WIP, which I agree, there are a few things to do, so @aleacevedo approval should be ignored:
> 
> * NODE_ENV Environment should determine the cached hostname, as currently defaults to `dev`
> * Documentation needs to be updated, this is mandatory as a user should be able to understand what `localhost:3000/dev` means, regardless of their knowledge about kyashu, capjs, etc
> * Having the documentation/guide on that, will allow others to test this correctly (caching) before approving
> * Noticed the change on Promise `createActor`, as it's not required, after I have refactored the initial proposed version into this latest class structure moved the signature which declared a promised, which you spotted as not necessary. This means that it should be refactored, as that initialisation was written to accommodate that and can now be simplified. This is probably best to do separately, let know if I should work on that as I'm integrating cap-js in cap-explorer might be quicker

Dev docs, ToDo 02aa679

File: src/index.ts [hunk 2]
```diff
@@ -125,17 +135,9 @@ export class CapBase<T> {
       });
     }
 
-    const router = "router" in args ? args.router : await CapRouter.init({ host: args.routerHost, canisterId: args.routerCanisterId });
```
Comment (b0xtch) [inline src/index.ts:128]: @aleacevedo check this out, I removed this piece. Create actor should not be async

File: src/index.ts [hunk 3]
```diff
@@ -125,17 +135,9 @@ export class CapBase<T> {
...
-
-    const { canister } = await router.get_token_contract_root_bucket({
-      tokenId: Principal.fromText(args.tokenId),
-    });
-
-    if (!canister?.[0]) throw Error(`Token ${args.tokenId} not in cap`);
-
     return Actor.createActor(idlFactory, {
       agent,
-      canisterId: canister[0],
```
Comment (aleacevedo) [inline src/index.ts:138]: You need al this to make possible create the actor from tokenId directly

File: src/index.ts [hunk 4]
```diff
@@ -125,17 +135,9 @@ export class CapBase<T> {
...
+      canisterId: args.tokenId,
```
Comment (aleacevedo) [inline src/index.ts:140]: It not gonna work. Here you cannot provide tokenId directly you have to pass the rootTokenId in other case u gonna create an actor from token

File: src/index.ts (hunk 2 above)
Comment (aleacevedo) [reply src/index.ts:128]: Why create actor should not be async?

File: src/index.ts (hunk 4 above)
Comment (aleacevedo) [reply src/index.ts:140]: You might do it here https://github.com/Psychedelic/cap-js/pull/6/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R149-R153 before call `createActor`, so you could make  createActor sync but you should keep `initialize` async. If it is not possible you should revert this PR: https://github.com/Psychedelic/cap-js/pull/9/files

**Review by aleacevedo — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (b0xtch) [reply src/index.ts:128]:
```
    static createActor<T = Record<string, ActorMethod>>(interfaceFactory: IDL.InterfaceFactory, configuration: ActorConfig): ActorSubclass<T>;

```

This is the definition of createActor in agent-js. We are promisifying something that doesn't need to be. Also with making it async we sacrifice having to createActors in class constructors (can't be async)

File: src/index.ts (hunk 4 above)
Comment (b0xtch) [reply src/index.ts]: 6e4b7d1 I accommodated both scenarios

**Review by aleacevedo — APPROVED** [review]

**Pull request merged**